pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -482,59 +481,35 @@ func (r *ReconcilePlanExecution) Reconcile(request reconcile.Request) (reconcile
 					log.Printf("PlanExecutionController: Cleanup failed: %v", err)
 				}
 
-				arg := obj.DeepCopyObject()
-				result, err := controllerutil.CreateOrUpdate(context.TODO(), r.Client, arg, func(newObj runtime.Object) error {
-					// TODO: Clean this up. I don't like having to do a switch here.
-					switch t := newObj.(type) {
-					case *appsv1.StatefulSet:
-						log.Printf("PlanExecutionController: CreateOrUpdate: StatefulSet %+v", t.Name)
-
-						newSs := newObj.(*appsv1.StatefulSet)
-						ss, ok := obj.(*appsv1.StatefulSet)
-						if !ok {
-							return fmt.Errorf("object passed in doesn't match expected StatefulSet type")
-						}
-
-						// We need some specialized logic in there.
-						//
-						// We can't just copy the Spec since there are other values like
-						// spec.updateState, spec.volumeClaimTemplates, etc. that are all generated
-						// from the object by the k8s controller.
-						//
-						// We just want to update things we can change.
-						newSs.Spec.Replicas = ss.Spec.Replicas
-
-						return nil
-					case *appsv1.Deployment:
-						newD := newObj.(*appsv1.Deployment)
-						d, ok := obj.(*appsv1.Deployment)
-						if !ok {
-							return fmt.Errorf("object passed in doesn't match expected deployment type")
-						}
-						newD.Spec.Replicas = d.Spec.Replicas
-						return nil
-					case *v1beta1.Deployment:
-						newD := newObj.(*v1beta1.Deployment)
-						d, ok := obj.(*v1beta1.Deployment)
-						if !ok {
-							return fmt.Errorf("object passed in doesn't match expected deployment type")
-						}
-						newD.Spec.Replicas = d.Spec.Replicas
-						return nil
-
-					case *batchv1.Job:
-					case *kudov1alpha1.Instance:
-					// Unless we build logic for what a healthy object is, assume its healthy when created.
-					default:
-						log.Print("PlanExecutionController: CreateOrUpdate: Type is not implemented yet")
-						return nil
+				//See if its present
+				key, _ := client.ObjectKeyFromObject(obj)
```

Comment:
what happens if it's not present? this seems worth checking, even if we continue on (is it worth recording an event to help debug?)

Reply:
There's really no risk of this since we've already used this object as a runtime.Object in above lines.